products/identityplatform/api.yaml

```diff
@@ -0,0 +1,408 @@
+        required: true
+      - !ruby/object:Api::Type::String
+        name: 'clientSecret'
+        description: |
+          OAuth client secret
+        required: true
+      - !ruby/object:Api::Type::Boolean
+        name: 'enabled'
+        description: |
+          If this IDP allows the user to sign in
+  - !ruby/object:Api::Resource
+    name: 'TenantDefaultSupportedIdpConfig'
+    base_url: 'projects/{{project}}/tenants/{{tenant}}/defaultSupportedIdpConfigs'
+    self_link: 'projects/{{project}}/tenants/{{tenant}}/defaultSupportedIdpConfigs/{{client_id}}'
+    create_url: 'projects/{{project}}/tenants/{{tenant}}/defaultSupportedIdpConfigs?idpId={{client_id}}'
+    update_verb: :PATCH
+    update_mask: true
+    description: |
+      Configurations options for the tenant for authenticating with a the standard set of Identity Toolkit-trusted IDPs.
+
+      You must enable the
+      ((Google Identity Platform)[https://console.cloud.google.com/marketplace/details/google-cloud-platform/customer-identity])
+      in the marketplace prior to using this resource.
+    properties:
+      - !ruby/object:Api::Type::String
+        name: 'name'
+        output: true
+        description: |
+          The name of the default supported IDP config resource
+      - !ruby/object:Api::Type::String
+        name: 'tenant'
+        required: true
+        url_param_only: true
+        description: |
+          The name of the tenant where this DefaultSupportedIdpConfig resource exists
+      - !ruby/object:Api::Type::String
+        name: 'clientId'
+        description: |
+          OAuth client ID
+        required: true
+      - !ruby/object:Api::Type::String
+        name: 'clientSecret'
+        description: |
+          OAuth client secret
+        required: true
+      - !ruby/object:Api::Type::Boolean
+        name: 'enabled'
+        description: |
+          If this IDP allows the user to sign in
+  - !ruby/object:Api::Resource
+    name: 'InboundSamlConfig'
+    base_url: 'projects/{{project}}/inboundSamlConfigs'
+    self_link: 'projects/{{project}}/inboundSamlConfigs/{{name}}'
+    create_url: 'projects/{{project}}/inboundSamlConfigs?inboundSamlConfigId={{name}}'
+    update_verb: :PATCH
+    update_mask: true
+    description: |
+      Inbound SAML configuration for a Identity Toolkit project.
+
+      You must enable the
+      ((Google Identity Platform)[https://console.cloud.google.com/marketplace/details/google-cloud-platform/customer-identity])
+      in the marketplace prior to using this resource.
+    properties:
+      - !ruby/object:Api::Type::String
+        name: 'name'
+        description: |
+          The name of the InboundSamlConfig resource. Must start with 'saml.' and can only have alphanumeric characters,
+          hyphens, underscores or periods. The part after 'saml.' must also start with a lowercase letter, end with an
+          alphanumeric character, and have at least 2 characters.
+        required: true
```

Comment:
The docs for `name` say `Ignored during create requests` and that they're of the projects/x/thing/id form. Are they just blatantly wrong?

Reply:
I've added a `name_from_self_link` flattener to this field via `terraform.yaml` so the form will be more usable.

I believe the `name` in the body is ignored in the create request, but the shortened name is used in the `create_url`. It might technically be a `url_param_only`, but that prevents reading

It should be `input` though, I'll add that